.github/workflows/changeset-check.yml

```
@@ -0,0 +1,115 @@
+name: Changeset Check
+
+on:
+  pull_request:
+    branches:
+      - develop
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - labeled
+      - unlabeled
+
+permissions:
+  contents: read
+  pull-requests: read
+
+jobs:
+  check-changeset:
+    name: Validate Changeset Required
+    runs-on: token-studio-linux-large
+
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@0634a2670c59f64b4a01f0f96f84700a4088b9f0 # v2.12.0
+        with:
+          egress-policy: audit
+
+      - name: Checkout repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          fetch-depth: 0
+          # Ensure we have the develop branch for comparison
+          ref: ${{ github.head_ref }}
+
+      - name: Fetch develop branch
+        run: |
+          git fetch origin develop:develop || git fetch origin main:develop || true
+          git branch -a
+
+      - name: Setup NodeJS Environment
+        uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020 # v4.4.0
+        with:
+          node-version: 22.x
```

```suggestion
          node-version: 22.20.0
```
use explicit version